tests/std/tests/GH_000140_adl_proof_construction/test.compile.pass.cpp

```diff
@@ -145,6 +146,25 @@ void test_promise() {
     promise<validator&>{allocator_arg, adl_proof_allocator<unsigned char>{}};
 }
 
+void test_valarray() {
+    using validator_class = holder<validator>;
+
+    valarray<validator_class> valarr1(42);
+
+    validator_class a[1]{};
+    valarray<validator_class> valarr2(a, 1);
+    valarr2.resize(172, a[0]);
+
+    valarray<validator_class> valarr3(a[0], 1);
+    valarr3 = valarr2[slice{0, 1, 1}];
+
+    auto valarr4 = valarr1;
+    valarr4      = valarr1;
+
+    auto valarr5 = static_cast<valarray<validator_class>&&>(valarr2);
+    valarr5      = static_cast<valarray<validator_class>&&>(valarr3);

```

Comment:
I think it would be more conventional to `#include <utility>` and use `std::move` here.

Reply:
I replaced some pre-existing casts with `std::forward` calls as well. Note that the `forward` and `move` calls are fully qualified despite `using namespace std` because this is an ADL test.